salt/modules/dpkg.py

```diff
@@ -335,17 +335,23 @@ def _get_pkg_license(pkg):
     return ", ".join(sorted(licenses))
 
 
-def _get_pkg_install_time(pkg):
+def _get_pkg_install_time(pkg, arch=None):
     '''
     Return package install time, based on the /var/lib/dpkg/info/<package>.list
 
     :return:
     '''
     iso_time = None
     if pkg is not None:
-        location = "/var/lib/dpkg/info/{0}.list".format(pkg)
-        if os.path.exists(location):
-            iso_time = datetime.datetime.utcfromtimestamp(int(os.path.getmtime(location))).isoformat() + "Z"
+        if arch is not None and arch != 'all':
+            location = "/var/lib/dpkg/info/{0}:{1}.list".format(pkg, arch)
+            if os.path.exists(location):
+                iso_time = datetime.datetime.utcfromtimestamp(int(os.path.getmtime(location))).isoformat() + "Z"
+
+        if iso_time is None:
+            location = "/var/lib/dpkg/info/{0}.list".format(pkg)
+            if os.path.exists(location):
+                iso_time = datetime.datetime.utcfromtimestamp(int(os.path.getmtime(location))).isoformat() + "Z"
```

Comment:
There are few issues here as I see it. First, the `iso_time` will be not None only if previous location exists. Therefore I would not repeat myself twice. Second, you do not need to call OS's stat three (!) times. Third, there is no logging involved, so if package installation time is not popping up while asked, we have no idea why. Also you repeating constants, which might be also needed to be fixed.

I would write this part this way instead:

```python
loc_root = '/var/lib/dpkg/info'
locations = []
if arch is not None and arch != 'all':
    locations.append(os.path.join(loc_root, '/{0}:{1}.list'.format(pkg, arch))
locations.append(os.path.join(loc_root, '/{}.list'.format(pkg))  # lower prio, if arch
# ... add here other locations on different Debian flavours, if any
for location in locations:
    try:
        iso_time = datetime.datetime.utcfromtimestamp(
            int(os.path.getmtime(location))).isoformat() + "Z"
        break
    except FileNotFoundError as err:
        log.debug('Unable to get package installation time: %s', err)
```

Also I am not sure if ISO format is properly generated here. I'd rather use time zone, something like (only `try/except` block):
```python
try:
    iso_time = datetime.datetime.utcfromtimestamp(
        int(os.path.getmtime(location))).replace(tzinfo=datetime.timezone.utc).isoformat()
except FileNotFoundError as err:
    log.debug('Unable to get package installation time: %s', err)
```

Reply:
.

Reply:
Even without testing your suggestion yet, for all packages with architecture ```!= 'all'```...in worst case, you log an error message, although you are able to retrieve the installation date correctly (from the other filename).

But I'll try to correct this and test again....